plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -168,22 +168,22 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 		sql_query += query_addon
 
 		if p.Query[i].Version <= db_version {
-			rows, err := db.Query(sql_query)
+			rows, err := p.DB.Query(sql_query)
```

Comment:
No, this isn't handled by the sql package.

I don't see how `maxlifetime` has any bearing on this problem.

`db.Ping()` is also not a good way to solve it. First you'd have to start a transaction to ensure that the query uses the same connection the ping just used. Then the ping would also just introduce extra latency as you're adding an additional query. Basically this is not what `Ping()` is for. There are several discussions on the matter, such as [this one](https://github.com/go-sql-driver/mysql/issues/82) and [this one](https://github.com/golang/go/issues/7619).